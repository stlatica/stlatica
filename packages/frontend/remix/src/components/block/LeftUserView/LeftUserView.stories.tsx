import { LeftUserView } from "./LeftUserView";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "block/LeftUserView",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return (
    <div className="h-[70vh] w-[300px]">
      <LeftUserView userID="userid" />
    </div>
  );
};
