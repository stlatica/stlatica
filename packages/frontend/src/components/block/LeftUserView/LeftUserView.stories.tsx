import React from "react";

import { LeftUserView } from "./LeftUserView";

import type { Story, StoryDefault } from "@ladle/react";

export default {
  title: "components/block/LeftUserView",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return (
    <div className="h-[70vh] w-[300px]">
      <LeftUserView userID="userid" />
    </div>
  );
};
