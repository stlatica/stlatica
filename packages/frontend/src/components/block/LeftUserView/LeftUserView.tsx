import type React from "react";

import { container } from "./LeftUserView.css";
import { LeftUserViewTop } from "./LeftUserViewTop";

type LeftUserViewProps = {
  readonly userID: string;
  // children: React.ReactNode;
};

/**
 *
 */
export const LeftUserView: React.FC<LeftUserViewProps> = ({ userID }) => {
  return (
    <div className={container}>
      <LeftUserViewTop />
      <div className="text-xl">User Name Hereeeeeeeee</div>
      <div className="text-gray-400">@{userID}</div>
      <div className="flex justify-center ">
        <hr className="m-3 w-4/5 border-black" />
      </div>
      <div className="text-sm">
        ここに自己紹介をかきまああああああああああああああああああああああああああああああああああああす文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数
      </div>
    </div>
  );
};
