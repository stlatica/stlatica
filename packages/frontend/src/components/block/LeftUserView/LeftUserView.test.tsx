import { render } from "@testing-library/react";
import React from "react";
import { describe, expect, test } from "vitest";

import { Story1 } from "./LeftUserView.stories";


describe("snapshot", () => {
  test("snapshot", () => {
    const { asFragment } = render(<Story1 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
