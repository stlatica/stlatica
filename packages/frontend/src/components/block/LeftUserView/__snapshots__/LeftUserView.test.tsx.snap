// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-[70vh] w-[400px]"
  >
    <div
      class="h-full w-full bg-gray-200 p-3"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="grid h-[170px] grid-cols-10 "
        >
          <div
            class="col-span-2 "
          >
            <div
              class="h-full"
            >
              <a
                href="/"
              >
                <div
                  class="flex justify-center"
                >
                  <button
                    class="border-2 border-black"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="3em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="3em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                      />
                    </svg>
                  </button>
                </div>
              </a>
            </div>
          </div>
          <div
            class="relative col-span-4 "
          >
            <div
              class="absolute w-full border-2 border-black bg-red-100 pb-[100%]"
            />
          </div>
          <div
            class="col-span-4 m-2 flex flex-col"
          >
            <div
              class="flex w-full justify-center pb-3"
            >
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                  style="width: 8em;"
                  tabindex="0"
                  type="button"
                >
                  フォロー中
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="pb-2"
            />
            <div>
              9999万 Follows
            </div>
            <div>
              9999 Follers
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-xl"
      >
        User Name Hereeeeeeeee
      </div>
      <div
        class="text-gray-400"
      >
        @userid
      </div>
      <div
        class="flex justify-center "
      >
        <hr
          class="m-3 w-[80%] border-black"
        />
      </div>
      <div
        class="text-sm"
      >
        ここに自己紹介をかきまああああああああああああああああああああああああああああああああああああす文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数文字数
      </div>
    </div>
  </div>
</DocumentFragment>
`;
